### PR TITLE
Fix delete; clarify extension

### PR DESCRIPTION
**Pull request by mattvarghese**
* Deleting a resource should use a resource with just resourceType and id
* Extension examples updated to always have a reverse uri naming convention

### CODE REVIEW

**Review by ssmillard — COMMENTED** [review]
For extensions, Would it be possible to have a structure such as:	"extension": [
		{
			"uri": "com.example.timestamp",
			"value": "2017-11-27T22:13:25Z"
		},
		{
			"uri": "com.cds-hooks.sandbox.myextension-practitionerspecialty",
			"value": "gastroenterology"
		}]

**Comment by dennispatterson** [comment]
@ssmillard we had a bit of discussion about the extensions topic and I dug up https://github.com/cds-hooks/docs/issues/133 and https://github.com/cds-hooks/docs/issues/89, where there was a decision made to handle it this way (for both extensions and prefetch) vs the uri/value pairs mechanism.

**Comment by isaacvetter** [comment]
Related: http://hl7.org/fhir/us/davinci-crd/2019May/hooks.html#propose-alternate-request

The issue with:
1. Deleting a resource should use a resource with just resourceType and id

Is that the same element name (`resource`) using different data types between delete vs create/update is inelegent depending upon the dev stack. 

An alternative to the proposal would be to introduce a new element name (e.g. resourceLocation) to be used for deletes, which could also enable a transition path, and cds service could even provide both. 

If we make any changes, we should ensure that CRD stays in sync as well.

**Comment by isaacvetter** [comment]
@dennispatterson , @brynrhodes and I talked through this issue. 

Ultimately, it is awkward for developers to treat the same `resource` element as either a resource "location" (aka ProcedureRequest/123) or as a full resource. We believe that delete suggestions are not yet widely implemented. Specific to delete suggestions, from a developer perspective, constructing a shell resource withonly a resource type and id 

```
`resourceId` | CONDITIONAL | *string* | A relative reference to the relevant resource in the suggestion. SHOULD be provided for delete suggestions. 
`resource` | OPTIONAL | *object* | A FHIR resource. When the `type` attribute is `create`, the `resource` attribute SHALL contain a new FHIR resource to be created.  For `update`, this holds the updated resource in its entirety and not just the changed fields. Use of this field to communicate a string of a FHIR id for delete suggestions is DEPRECATED and `resourceId` SHOULD be used instead.
```
~~For `delete`, this SHALL be the id of the resource to remove. In hooks where only one "content" resource is ever relevant, this attribute MAY be omitted for `delete` action types only.~~

Todo:
- @isaacvetter : create jira issue

**Comment by isaacvetter** [comment]
Hey @ssmillard,

>For extensions, Would it be possible to have a structure such as: "extension": [

Yes, definitely. The idea is that everything within the extension element is entirely yours to do with as you want.

File: docs/specification/current.md [hunk 1]
```diff
@@ -503,7 +503,8 @@ Field | Optionality | Type | Description
...
 `description` | REQUIRED | *string* | Human-readable description of the suggested action MAY be presented to the end-user.
-`resource` | OPTIONAL | *object* | Depending upon the `type` attribute, a new resource or the id of a resource. When the `type` attribute is `create`, the `resource` attribute SHALL contain a new FHIR resource to be created. For `delete`, this SHALL be the id of the resource to remove. In hooks where only one "content" resource is ever relevant, this attribute MAY be omitted for `delete` action types only. For `update`, this holds the updated resource in its entirety and not just the changed fields.
+`resource` | CONDITIONAL | *object* | A FHIR resource. When the `type` attribute is `create`, the `resource` attribute SHALL contain a new FHIR resource to be created.  For `update`, this holds the updated resource in its entirety and not just the changed fields. Use of this field to communicate a string of a FHIR id for delete suggestions is DEPRECATED and `resourceId` SHOULD be used instead.
+`resourceId` | CONDITIONAL | *string* | A relative reference to the relevant resource in the suggestion. SHOULD be provided for when the `type` attribute is `delete`.
```
Comment (dennispatterson) [inline docs/specification/current.md:507]: Minor wordsmithing
```suggestion
`resourceId` | CONDITIONAL | *string* | A relative reference to the relevant resource. SHOULD be provided when the `type` attribute is `delete`. 
```

File: docs/specification/current.md [hunk 2]
```diff
@@ -539,7 +540,8 @@ The following example illustrates a delete action:
 {
 	"type": "delete",
 	"description": "Remove the inappropriate order",
-	"resource": "ProcedureRequest/procedure-request-1"
+	"resourceId": "ServiceRequest/procedure-request-1"
+	}
```
Comment (dennispatterson) [inline docs/specification/current.md:544]: Get rid of the extra bracket
```suggestion
```

**Review by dennispatterson — CHANGES_REQUESTED** [review]

**Comment by mattvarghese** [comment]
Hey Folks - are we all agreed on this change? Can we move ahead with this pull request?

**Review by dennispatterson — APPROVED** [review]

**Review by isaacvetter — APPROVED** [review]

**Pull request merged**